platform/ui-next/src/components/AllInOneMenu/Menu.tsx

```diff
@@ -108,6 +108,15 @@ const Menu = (props: MenuProps) => {
   ]);
   const [itemPanelLabels, setItemPanelLabels] = useState<Array<string>>([]);
 
+  // If the props change for the this top level menu then we have to update the menu path
+  // because the props to be rendered are maintained in the state.
+  useEffect(() => {
```

Comment:
The beauty of the AllInOnMenu is that it does not cascade out. So we keep all the menu that would otherwise cascade out on screen here. The downside is if the menu props change. This does the trick, but I would love to hear your feedback. 😊